queue_services/entity-emailer/src/entity_emailer/email_processors/special_resolution_notification.py

```diff
@@ -0,0 +1,97 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Email processing rules and actions for Special Resolution notifications."""
+from __future__ import annotations
+
+import re
+from pathlib import Path
+
+from entity_queue_common.service_utils import logger
+from flask import current_app
+from jinja2 import Template
+from legal_api.models import Filing, UserRoles
+
+from entity_emailer.email_processors import get_filing_info, get_recipient_from_auth, substitute_template_parts
+from entity_emailer.email_processors.special_resolution_helper import get_completed_pdfs, get_paid_pdfs
+
+
+def process(email_info: dict, token: str) -> dict:  # pylint: disable=too-many-locals, too-many-branches
+    """Build the email for Special Resolution notification."""
+    logger.debug('special_resolution_notification: %s', email_info)
+    # get template and fill in parts
+    filing_type, status = email_info['type'], email_info['option']
+    # get template vars from filing
+    filing, business, leg_tmz_filing_date, leg_tmz_effective_date = get_filing_info(email_info['filingId'])
+    filing_name = filing.filing_type[0].upper() + ' '.join(re.findall('[a-zA-Z][^A-Z]*', filing.filing_type[1:]))
+
+    template = Path(
+        f'{current_app.config.get("TEMPLATE_PATH")}/SR-CP-{status}.html'
+    ).read_text()
+    filled_template = substitute_template_parts(template)
+    # render template with vars
+    jnja_template = Template(filled_template, autoescape=True)
+    filing_data = (filing.json)['filing'][f'{filing_type}']
+    name_changed = filing.filing_json['filing'].get('changeOfName')
+    rules_changed = bool(filing.filing_json['filing'].get('restoration', {}).get('rulesFileKey'))
+    html_out = jnja_template.render(
+        business=business,
+        filing=filing_data,
+        header=(filing.json)['filing']['header'],
+        filing_date_time=leg_tmz_filing_date,
+        effective_date_time=leg_tmz_effective_date,
+        entity_dashboard_url=current_app.config.get('DASHBOARD_URL') +
+        (filing.json)['filing']['business'].get('identifier', ''),
+        email_header=filing_name.upper(),
+        filing_type=filing_type,
+        name_changed=name_changed,
+        rules_changed=rules_changed
+    )
+
+    # get attachments
+    if status == Filing.Status.PAID.value:
+        pdfs = get_paid_pdfs(token, business, filing, leg_tmz_filing_date, leg_tmz_effective_date)
+    if status == Filing.Status.COMPLETED.value:
+        pdfs = get_completed_pdfs(token, business, filing, name_changed)
+
+    # get recipients
+    identifier = filing.filing_json['filing']['business']['identifier']
+    recipients = []
+    recipients.append(get_recipient_from_auth(identifier, token))
```

Comment:
I think you might have an issue here.  From the SR email requirements, you need to email the registered office and the completing party.

The registered office email is what you are retrieving via `get_recipient_from_auth(identifier, token)` so I think we are good there.

But the completing party is never saved to the filing json even though the SR designs have it auto-populated.  i.e. we don't have the completing party to send to currently
![image](https://github.com/bcgov/lear/assets/6685223/bc9b0d09-8fe6-4cad-9f05-b2138b44c9e0)

I think the business schema for SR needs to be updated to include a `parties` block so the completing party can be added when the filing json is saved.  I'm assuming some FE work will also need to be done to ensure the completing party is added to the filing json.

Reply:
This should solve the issue https://github.com/bcgov/lear/blob/main/queue_services/entity-emailer/src/entity_emailer/email_processors/filing_notification.py#L276

Reply:
I just look at SR base filing, completing email send with ['filing']['header']['documentOptionalEmail'], I think recipients added as:
```
    if filing.submitter_roles and UserRoles.staff in filing.submitter_roles:
        recipients.append(filing.filing_json['filing']['header'].get('documentOptionalEmail'))
 ```
 I'll modify business-schema to include parties for both SR and SR correction

Reply:
Refer the above link. You don't need to modify schema

Reply:
@vysakh-menon-aot Thanks! I updated.